packages/flutter/lib/src/rendering/list_wheel_viewport.dart

```diff
@@ -828,22 +828,51 @@ class RenderListWheelViewport
 
   /// Paints all children visible in the current viewport.
   void _paintVisibleChildren(PaintingContext context, Offset offset) {
+    if (overAndUnderCenterOpacity >= 1) {
+      RenderBox? childToPaint = firstChild;
+      while (childToPaint != null) {
+        final ListWheelParentData childParentData = childToPaint.parentData! as ListWheelParentData;
+        // center value is ignored if overAndUnderCenterOpacity >= 1.
+        _paintTransformedChild(childToPaint, context, offset, childParentData.offset, center: null);
+        childToPaint = childAfter(childToPaint);
+      }
+      return;
+    }
+
+    // In order to reduce the number of opacity layers, we first paint all
+    // partially opaque children, then finally paint the fully opaque children.
+    context.pushOpacity(offset, (overAndUnderCenterOpacity * 255).round(), (PaintingContext context, Offset offset) {
+      RenderBox? childToPaint = firstChild;
+      while (childToPaint != null) {
+        final ListWheelParentData childParentData = childToPaint.parentData! as ListWheelParentData;
+        _paintTransformedChild(childToPaint, context, offset, childParentData.offset, center: false);
+        childToPaint = childAfter(childToPaint);
+      }
```

Comment:
Wondering if we can refactor this into a method (with `center` as a configurable argument) and call that here, above, and below to de-dup this a little and show that it is essentially doing the same thing?

Reply:
Done